new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -0,0 +1,1237 @@
+
+    // Check standalone server configuration is available 
+    isServerConfigured = 
+       checkManagedServerConfiguration("standalone-managed");
+    logger.info("Found non-cluster managed server configuration");
+  }
+
+  /**
+   * Stop the Administration server by patching the resource defintion with 
+   *  spec/adminServer/serverStartPolicy set to NEVER.
+   * Make sure that Only the Administration server is stopped. 
+   * Restart the Administration server by patching the resource defintion with 
+   *  spec/adminServer/serverStartPolicy set to IF_NEEDED.
+   * Make sure that Administration server is in RUNNING state. 
+   */
+  @Test
+  @DisplayName("Restart the Administration server with serverStartPolicy")
+  public void testAdminServerRestart() {
+
+    StringBuffer patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"NEVER\"")
+        .append(" }]");
+    logger.info("AdminServer shutdown patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(adminShutdown) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(adminShutdown) failed");
+   
+    checkPodDeleted(adminServerPodName, domainUid, domainNamespace);
+    logger.info("AdminServer shutdown success");
+
+    // check managed server pods are not affected
+    logger.info("Check managed server pods are not affected");
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+    logger.info("Managed servers are still RUNNING after AdminServer is stopped");
+    patchStr = null;
+    patchStr = new StringBuffer("[{");
+    patchStr.append("\"op\": \"replace\",")
+        .append(" \"path\": \"/spec/adminServer/serverStartPolicy\",")
+        .append(" \"value\":  \"IF_NEEDED\"")
+        .append(" }]");
+    logger.info("AdminServer restart patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    crdPatched = false;
+    crdPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restart) failed");
+
+    assertTrue(crdPatched, "patchDomainCustomResource(restart) failed");
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkServiceExists(adminServerPodName, domainNamespace);
```

Comment:
checkServiceExists has to be called before checkPodReady

Reply:
done